Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -0,0 +1,302 @@
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorGradientBakerComponent::PreviewSettingsAndSettingsVisibilityChanged)
+                    ->DataElement(0, &EditorGradientBakerComponent::m_previewPosition, "Preview Position", "Center of the preview bounds")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorGradientBakerComponent::UpdatePreviewSettings)
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &EditorGradientBakerComponent::GetPreviewPositionVisibility)
+                    ->DataElement(0, &EditorGradientBakerComponent::m_previewSize, "Preview Size", "Size of the preview bounds")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorGradientBakerComponent::UpdatePreviewSettings)
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &EditorGradientBakerComponent::GetPreviewSizeVisibility)
+                    ->DataElement(0, &EditorGradientBakerComponent::m_constrainToShape, "Constrain to Shape", "If checked, only renders the parts of the gradient inside the component's shape and not its entire bounding box")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorGradientBakerComponent::UpdatePreviewSettings)
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &EditorGradientBakerComponent::GetPreviewConstrainToShapeVisibility)
+                    ;
+            }
+        }
+    }
+
+    void EditorGradientBakerComponent::Activate()
+    {
+        AzToolsFramework::Components::EditorComponentBase::Activate();
+
+        m_gradientEntityId = GetEntityId();
+
+        SectorDataNotificationBus::Handler::BusConnect();
+        LmbrCentral::DependencyNotificationBus::Handler::BusConnect(GetEntityId());
+        AzToolsFramework::EntitySelectionEvents::Bus::Handler::BusConnect(GetEntityId());
+        GradientPreviewContextRequestBus::Handler::BusConnect(GetEntityId());
+
+        m_configuration.m_gradientSampler.m_ownerEntityId = GetEntityId();
+
+        // Validation needs to happen after the ownerEntity is set in case the validation needs that data
+        if (!m_configuration.m_gradientSampler.ValidateGradientEntityId())
+        {
+            SetDirty();
+        }
+
+        m_dependencyMonitor.Reset();
+        m_dependencyMonitor.ConnectOwner(GetEntityId());
+        m_dependencyMonitor.ConnectDependency(m_configuration.m_gradientSampler.m_gradientId);
+
+        // Connect to GradientRequestBus last so that everything is initialized before listening for gradient queries.
+        GradientRequestBus::Handler::BusConnect(GetEntityId());
+
+        if (!m_previewEntityId.IsValid())
+        {
+            // Default our preview entity to this entity
+            m_previewEntityId = GetEntityId();
+            SetDirty();
+        }
+
+        UpdatePreviewSettings();
```

Comment:
Suggestion: Maybe add a comment explaining why it's ok to update the preview stuff _after_ the "Connect to GradientRequestBus last" comment. It just seemed a little jarring on first read to see "connect last", then have more logic after that point.